databases/week1/database-lesson1.sql

```diff
@@ -0,0 +1,135 @@
+insert into task (id, title, description, created, updated, due_date, status_id, user_id) values (30, 'Remove facebook from phone', 'To avoid interruptions when working', '2017-10-26 17:15:07', '2017-10-13 03:36:47', '2017-12-19 11:10:02', 3, 4);
+insert into task (id, title, description, created, updated, due_date, status_id, user_id) values (31, 'Backup databases to external disk', 'Remember to store the disk in another physical location', '2017-09-09 17:32:33', '2017-10-01 21:18:59', '2017-12-23 14:21:01', 1, 2);
+insert into task (id, title, description, created, updated, due_date, status_id, user_id) values (32, 'Put up the new lamp in the hallway', NULL, '2017-10-15 05:45:54', '2017-10-16 14:05:35', '2017-12-29 02:29:26', 3, 3);
+insert into task (id, title, description, created, updated, due_date, status_id, user_id) values (33, 'Hang up paintings in living room', NULL, '2017-09-10 05:36:11', '2017-10-09 17:40:42', null, 3, 4);
+insert into task (id, title, description, created, updated, due_date, status_id, user_id) values (34, 'Buy plane ticket to Auckland', 'Check prices online first!', '2017-09-05 09:07:22', '2017-10-15 09:36:06', '2017-12-07 11:10:05', 1, 9);
+insert into task (id, title, description, created, updated, due_date, status_id, user_id) values (35, 'Learn about NoSQL databases', 'MongoDB, CouchDB, etc.', '2017-10-20 01:41:53', '2017-10-04 07:19:56', '2017-12-23 10:13:42', 2, NULL);
+
+
+--HOMEWORK
+
+-- Select the names and phones of all users;
+SELECT name, phone FROM user;
+
+-- Select the name of the user with id=10;
+SELECT name FROM user
+WHERE id=10;
+
+-- Find how many users exist in the database;
+SELECT count(*) FROM user;
+
+-- Select the names of the first 5 users in the database;
+SELECT name FROM user LIMIT 5;
+
+-- Select the names of the last 3 users in the database;
+SELECT name FROM user ORDER BY name DESC LIMIT 3;
+
+-- Sum all the ids in the user table;
+SELECT SUM(id) FROM user;
+
+-- Select all users and order them alphabetically by name;
+SELECT * FROM user ORDER BY name ASC;
+
+-- Find all tasks that include SQL either on the title or on the description;
+SELECT title FROM task 
+WHERE title LIKE '%SQL%' OR description LIKE '%SQL%';
+
+-- Find the title of all tasks that the user Maryrose is responsible for;
+SELECT * FROM task;
+SELECT * FROM user;
+
+SELECT task.title, user.id
+FROM  task INNER JOIN user ON task.user_id = user.id
+WHERE user.name LIKE '%Maryrose%';
+
+-- Find how many tasks each user is responsible for;
+-- !!!This query doesn't work even though I think I wrote it correctly !!!
+SELECT user.name, COUNT(*) AS task_count
+FROM  task JOIN user ON task.user_id = user.id GROUP BY user.id;
+
+-- Find how many tasks with a status=Done each user is responsible for;
+SELECT COUNT(*) AS task_count
+FROM  task JOIN status ON task.status_id = status.id
+WHERE status.id = '3';
```

Comment:
You should've found how many tasks each user has with status Done:
SELECT task.user_id, COUNT(*) AS task_count
FROM  task 
JOIN status ON task.status_id = status.id
WHERE status.id = '3'
GROUP BY task.user_id;